### PR TITLE
Drop support for Mojang accounts

### PR DESCRIPTION
**Pull request by Trial97**
The Mojang login is no longer supported as of this month.

### CODE REVIEW

**Review by ColonelGerdauf — APPROVED** [review]

**Review by getchoo — APPROVED** [review]
lgtm! after getting rid of the last remaining mentions of mojang accounts being supported that i could find, the only suggestion i have is removing/disabling the account type column in the account list. it seems a bit pointless now with only msa being supported

**Comment by Trial97** [comment]
we also support offline account type there so I vote to keep it.

**Comment by ColonelGerdauf** [comment]
offline account type is quite useful for versions 1.5.2 and earlier, when the original Mojang launcher was not locked behind a login. Not to mention, without Koad's skin fix, there was no use for the accounts anyways.

And even for later versions, though I am not sure how well some of the components would work.

**Comment by getchoo** [comment]
> we also support offline account type there so I vote to keep it.

oops! i forgot about that for a second haha

File: launcher/LaunchController.cpp [hunk 1]
```diff
@@ -88,8 +88,8 @@ void LaunchController::decideAccount()
     if (accounts->count() <= 0) {
         // Tell the user they need to log in at least one account in order to play.
         auto reply = CustomMessageBox::selectable(m_parentWidget, tr("No Accounts"),
-                                                  tr("In order to play Minecraft, you must have at least one Microsoft or Mojang "
-                                                     "account logged in. Mojang accounts can only be used offline. "
+                                                  tr("In order to play Minecraft, you must have at least one Microsoft "
+                                                     "account logged in."
```
Comment (TheKodeToad) [inline launcher/LaunchController.cpp]:
```suggestion
                                                     "account which owns Minecraft logged in."
```

**Review by TheKodeToad — CHANGES_REQUESTED** [review]
might as well, right?

File: launcher/minecraft/auth/AccountList.cpp [hunk 2]
```diff
@@ -379,11 +366,9 @@ QVariant AccountList::headerData(int section, [[maybe_unused]] Qt::Orientation o
                 case NameColumn:
                     return tr("User name of the account.");
                 case TypeColumn:
-                    return tr("Type of the account - Mojang or MSA.");
+                    return tr("Type of the account (currently only MSA is supported.)");
```
Comment (TheKodeToad) [inline launcher/minecraft/auth/AccountList.cpp]:
```suggestion
                    return tr("Type of the account - Msa or Offline.");
```

**Review by Scrumplex — APPROVED** [review]

File: launcher/minecraft/auth/AccountList.cpp (hunk 2 above)
Comment (Scrumplex) [reply launcher/minecraft/auth/AccountList.cpp]:
```suggestion
                    return tr("Type of the account (MSA or Offline)");
```

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**